src/types/constant.ts

```diff
@@ -0,0 +1,26 @@
+import {isFactory, IFactory} from "../core/factories"
+import {invariant, fail, isPrimitive} from "../utils"
+import {PrimitiveType} from "./primitive"
+
+export class Constant extends PrimitiveType {
+    readonly value: any
+
+    constructor(value: any) {
+        super("" + value)
+        this.value = value
+    }
+
+    describe(){
+        return this.value
```

Comment:
This is done to make typescript-like describes. TypeScript itself shows "a" | "b" | "c", so maybe it should be `JSON.stringify(this.value)`